projects/noloan/admin/src/main/java/hasadna/noloan/admin/app/SuggestionRecyclerAdapter.java

```diff
@@ -2,45 +2,76 @@
 
 import android.os.Handler;
 import android.os.Looper;
+
 import androidx.annotation.NonNull;
 import androidx.recyclerview.widget.RecyclerView;
+
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.Button;
 import android.widget.TextView;
 import android.widget.Toast;
 
+import java.util.ArrayList;
+
 import hasadna.noloan.common.FirestoreClient;
 import hasadna.noloan.common.SmsMessages;
 import hasadna.noloan.protobuf.SmsProto.SmsMessage;
 
-public class SuggetionRecyclerAdapter
-    extends RecyclerView.Adapter<SuggetionRecyclerAdapter.RecyclerViewHolder> {
+public class SuggestionRecyclerAdapter
+    extends RecyclerView.Adapter<SuggestionRecyclerAdapter.RecyclerViewHolder> {
+
+  ArrayList<SmsMessage> messages;
 
-  SmsMessages DbMessages;
+  public SuggestionRecyclerAdapter() {
+    messages = new ArrayList<>();
+    SmsMessages dbMessages = SmsMessages.get();
```

Comment:
IMO, this is a little confusing. I think if you need to save it as a local variable, and what you are saving is just the `SmsMessages` instance, as in - you aren't saving the actual dbMessages, then rather name it `smsMessages`

Personally if this local variable is only for the usage of constructing the `SuggestionRecyclerAdapter` and it isn't a class member, I'd just go with `SmsMessages.get()` for this purpose. Not sure how much it matters though, and probably a matter of style.

Reply:
It is used in 2 places so I cached it.
I don't know what it is confusing to you, sorry

Reply:
It is used twice so I cached it.
I am sorry it confusing for you...

Reply:
Yeah, I got that... A matter of style I suppose....
Cool :+1: